dev/archery/archery/crossbow/cli.py

```diff
@@ -558,11 +558,14 @@ def upload_artifacts(obj, tag, sha, patterns, method):
               help='Just display process, don\'t download anything')
 @click.option('--days', default=90,
               help='Branches older than this amount of days will be deleted')
+@click.option('--maximum', default=1000,
+              help='Maximum limit of branches to delete for a single run')
 @click.pass_obj
-def delete_old_branches(obj, dry_run, days):
+def delete_old_branches(obj, dry_run, days, maximum):
     """
     Deletes branches on queue repository (crossbow) that are older than number
     of days.
+    With a maximum number of branches to be deleted.
```

Comment:
Should we add a note like "deleting to many branches at once can trigger the Github abuse protection"?

Reply:
I've added a comment about why is required.